### PR TITLE
Use objdir as built library path

### PR DESCRIPTION
**Pull request by mihails-strasuns**
DMD `-od` flag defines where to write object & library files, thus for library
rules result won't be in `bindir` but `objdir`.

### CODE REVIEW

**Comment by mihails-strasuns** [comment]
Hm, will require updating `objdir` field too, one moment

**Comment by mihails-strasuns** [comment]
Ok, in the end it had to be quite a bit more complicated - original assumption that `output` can be calculated in uniform manner had to be adjusted in several places.

File: scripts/rules/d/dmd.lua [hunk 1]
```diff
@@ -223,12 +223,16 @@ function common:rules()
         rule {
             inputs = table.join(objs, libs),
             task = {table.join(args, linker_opts, objs, libs)},
-            outputs = table.join({output}),
+            outputs = {self.path()},
```
Comment (jasonwhite) [inline scripts/rules/d/dmd.lua]: I think this should be `self:path()`.

Comment (mihails-strasuns) [reply scripts/rules/d/dmd.lua]: Correct. Updated.

**Comment by mihails-strasuns** [comment]
Thanks!

**Pull request merged**